install_gridpack_deps.sh

```diff
@@ -1,133 +1,168 @@
-# This script installs all GridPACK dependencies.The dependencies are installed in external-dependencies directory.
-
-# This script should be run from the top-level GridPACK directory.
-
-# Flags for installing/not installing different dependency packages
-install_boost=true
-install_ga=true
-install_petsc=true
-
-echo $(date)
-
-if test -z ${GRIDPACK_ROOT_DIR}
-then
-    export GRIDPACK_ROOT_DIR=${PWD}
-    echo "GRIDPACK_ROOT_DIR = ${GRIDPACK_ROOT_DIR}"
-fi
-
-cd ${GRIDPACK_ROOT_DIR}
-
-# Create directory for installing external packages
-if test -z ${GP_EXT_DEPS}
-then
-     export GP_EXT_DEPS=${GRIDPACK_ROOT_DIR}/external-dependencies
-     rm -rf ${GP_EXT_DEPS}
-     mkdir ${GP_EXT_DEPS}
-     echo "GRIDPACK_EXT_DEPENDENCIES_DIR=${GP_EXT_DEPS}"
-else
-    if test -d ${GP_EXT_DEPS}
-    then
-	echo "GRIDPACK_EXT_DEPENDENCIES_DIR=${GP_EXT_DEPS}"
-    else
-	mkdir ${GP_EXT_DEPS}
-    fi		
-fi
-
-cd ${GP_EXT_DEPS}
-
-if ${install_boost}
-then
+#! /bin/bash
 
-  rm -rf boost*
-    
-  # Download and install Boost
-  echo "Downloading Boost-1.78.0"
+# installs GridPACK dependencies to the current directory
+
+set -xeuo pipefail
+
+function install_boost {
+  local boost_version=$1
 
-  # Download Boost
-  wget https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.tar.gz
+  # check args
+  : "${boost_version:?}"
 
-  # Untar
-  tar -xf boost_1_78_0.tar.gz
+  echo "--- Installing Boost ${boost_version} ---"
+
+  # remove existing
+  rm -rf boost*
 
-  cd boost_1_78_0
+  # download
+  echo "Downloading Boost"
+  wget \
+    "https://boostorg.jfrog.io/artifactory/main/release/${boost_version}/source/boost_${boost_version//./_}.tar.gz" \
+    -O boost.tar.gz \
+    --quiet
 
-  # Install boost
-  echo "Building Boost-1.78.0"
+  # unpack
+  echo "Unpacking Boost"
+  tar -xf boost.tar.gz && rm boost.tar.gz
 
-  ./bootstrap.sh --prefix=install_for_gridpack --with-libraries=mpi,serialization,random,filesystem,system
+  # remove version from dir name
+  mv "boost_${boost_version//./_}" boost
 
-  echo 'using mpi ;' >> project-config.jam
+  pushd boost || exit
 
+  # bootstrap
+  echo "Bootstrapping Boost"
+  ./bootstrap.sh \
+    --prefix=install_for_gridpack \
+    --with-libraries=mpi,serialization,random,filesystem,system
+  echo 'using mpi ;' >>project-config.jam
```

Comment:
For RHEL/Rocky, the `openmpi` module needs to be loaded before this.

Reply:
Fixed